proxy-node-shared/src/main/java/uk/gov/ida/notification/validations/Base64SamlValidator.java

```diff
@@ -0,0 +1,36 @@
+package uk.gov.ida.notification.validations;
+
+import org.apache.commons.lang3.StringUtils;
+import org.glassfish.jersey.internal.util.Base64;
+import org.opensaml.saml.saml2.core.impl.ResponseImpl;
+import uk.gov.ida.notification.saml.SamlParser;
+
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+
+public class Base64SamlValidator implements ConstraintValidator<ValidBase64Saml, String> {
+
+    public static final int MinLength = 64;
+    public static final int MaxLength = 64000;
+
+    @Override
+    public void initialize(ValidBase64Saml constraint) { /* intentionally blank */ }
+
+    @Override
+    public boolean isValid(String value, ConstraintValidatorContext context) {
+        if (StringUtils.isBlank(value)) { return true; } // @NotNull should detect nulls
+
+        try {
+            String decoded = Base64.decodeAsString(value);
+            if (StringUtils.isBlank(decoded)) {
+                return false;
+            }
+
+            Object saml = new SamlParser().parseSamlString(decoded);
```

Comment:
Can we be more specific than just object?

Reply:
Sadly, no. If we were going to do that, we'd write a validator for each possible SAML message returned - but we're not validating the content of the SAML here, we're validating that the parameter passed to us contains _some_ SAML.

Reply:
On further inspection, `parseSamlString` is a generic method, and I'd need to provide it with a concrete type. I could possibly use `ResponseImpl`. I'm not sure it adds much beyond a bit of casting. What do you think?

Reply:
(done)